src/CodeExplorerBundle/Twig/SourceCodeExtension.php

```diff
@@ -41,13 +41,16 @@ public function setController($controller)
     public function getFunctions()
     {
         return array(
-            new \Twig_SimpleFunction('show_source_code', array($this, 'showSourceCode'), array('is_safe' => array('html'), 'needs_environment' => true)),
+            new \Twig_SimpleFunction('show_source_code', array($this, 'showSourceCode'), array(
+                'is_safe' => array('html'),
+                'node_class' => 'CodeExplorerBundle\Twig\Node\TemplateAwareFunction'
+            )),
         );
     }
 
-    public function showSourceCode(\Twig_Environment $twig, \Twig_Template $template)
+    public function showSourceCode(\Twig_Template $template)
     {
-        return $twig->render('@CodeExplorer/source_code.html.twig', array(
+        return $template->getEnvironment()->render('@CodeExplorer/source_code.html.twig', array(
```

Comment:
@stof, I can pass the `Twig_Environment` as a second argument:
```php
$compiler->raw('($this, $this->env)');
```
Supporting the existing features would require to copy all the code from the `compileArguments` method https://github.com/twigphp/Twig/blob/1.x/lib/Twig/Node/Expression/Call.php#L37. Is it really necessary?